accepted/00080-api-versioning.md

```diff
@@ -0,0 +1,295 @@
+  this doesn't need to be true.
+- API Consumers would need to introduce a check for the flavor in
+  their scripts.
+
+#### Pros
+- Trivial implementation on server side
+
+#### Cons
+- Tracking non-breaking changes between various SUMA maintenance
+  updates is a bit more complex (need to use the API introspection
+  calls, e.g. `getApiCallList`).
+- Brings complexity to the API consumers and `spacecmd`. The scripts
+  would need to check the flavor and the API version.
+
+
+### Solution 2: No flavor, use `systemVersion` instead
+
+Based on the [Solution 1](#solution-1)
+
+In this case, the API must be stable within a Uyuni/SUMA minor
+version. (TODO @hustodemon: ask @moio/@mc, but i think this should be
+the case already).
+
+The user scripts could then use the `api.systemVersion` instead of
+checking the flavor and API version.
+
+#### Pros
+- No implementation needed on the server side
+
+#### Cons
+- Version format: the scripts need to consider various corner cases
+  (`4.2.0 RC2`, `2021.05`) and would need to make sure they are
+  handled correctly by implementing various, possibly error-prone
+  comparators in their code.
+- Same problem with tracking non-breaking changes like in the
+  [Solution 1](#solution-1).
+
+
+### Solution 3: Bump API version on any change in the API
+
+Based on the [Solution 1](#solution-1)
+
+- Same as Solution 1, but the version gets bumped on any (even
+non-breaking) change within a product release or a maintenance update.
+- Brekaing API within a SUMA maintenance update is still forbidden.
+
+#### Pros & Cons
+- Trivial implementation on the server side
+- Does not suffer from the first "Cons" of the [Solution 1](#solution-1)
+
+#### Cons
+- Brings complexity to the API consumers and `spacecmd`. The scripts
+  would need to check the flavor and the API version.
+
+
+### Solution 4: Use a `major.minor` versioning scheme
+
+Currently, the API version is an integer. This solution uses 2
+integers (or a list of integers in general). The first number
+describes the major version, the second one describes the minor
+version.
+
+Bump the major version with the product release, bump the minor one on
+non-breaking changes.
+
+The version bumping is described by this example:
+- After SUMA 4.2 release (API version is `25`) Uyuni bumps the version
+  to `26.01`
+- On SUMA 4.3 release, the SUMA API version gets set to the current
+  Uyuni API version at that time, e.g `26.10`
+- Uyuni API version gets increased to 27.01
+- On backporting features from Uyuni which change the API, SUMA 4.3
+  increases the minor version `26.11`, `26.12`, etc...
+
+
+#### Pros
+- Consumers able to track non-breaking changes easily (same as
+  [Solution 3](#solution-3-bump-api-version-on-any-change-in-the-api))
+
+#### Cons
+- Frequent changes in minor version could mean more burden on
+  developers and release engineers. A [CI job](#ci-job) needs to be
+  present to ease this.
```

Comment:
Usage on the client side is also not easy. As a special feature/function could be available in 26.13 and 27.03.
A simple >= 26.13 will also match versions 27.01 and 27.02 which do not have this feature.

Reply:
@hustodemon how would you return such a version? Non exhaustive list of possibilities: as a string, as a floating point number, as a struct, via two `getMinor()`/`getMajor()` methods...

Reply:
Sounds good, please copypaste those observations in the text for the posterity!

Reply:
In the end I decided not to go with the major/minor scheme. We can revive it when necessary.